module3/homework/passwordCheck/validation.cpp

```diff
@@ -1,2 +1,72 @@
+#include <algorithm>
+#include <ostream>
+#include <sstream>
+#include <string>
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+
+std::ostream& operator<<(std::ostream& os, ErrorCode value)
+{
+    switch (value)
+    {
+        case ErrorCode::Ok:
+            return os << "OK";
+        case ErrorCode::PasswordNeedsAtLeastNineCharacters:
+            return os << "PasswordNeedsAtLeastNineCharacters";
+        case ErrorCode::PasswordNeedsAtLeastOneNumber:
+            return os << "PasswordNeedsAtLeastOneNumber";
+        case ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter:
+            return os << "PasswordNeedsAtLeastOneSpecialCharacter";
+        case ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter:
+            return os << "PasswordNeedsAtLeastOneUppercaseLetter";
+        case ErrorCode::PasswordsDoesNotMatch:
+            return os << "PasswordsDoesNotMatch";
+    }
+    return os;
+}
```

Comment:
Why did you use operator << overloading?

In my opinion the simplies way is:
```suggestion
std::string getErrorMessage(const ErrorCode value)
{
    switch (value)
    {
        case ErrorCode::Ok:
            return "OK";
        case ErrorCode::PasswordNeedsAtLeastNineCharacters:
            return "PasswordNeedsAtLeastNineCharacters";
        case ErrorCode::PasswordNeedsAtLeastOneNumber:
            return "PasswordNeedsAtLeastOneNumber";
        case ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter:
            return "PasswordNeedsAtLeastOneSpecialCharacter";
        case ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter:
            return "PasswordNeedsAtLeastOneUppercaseLetter";
        case ErrorCode::PasswordsDoesNotMatch:
            return "PasswordsDoesNotMatch";
    }

}
```

Reply:
with overloaded operator enums are visible in gtest log when enums does not match.